### DISCUSSION TITLE
Documentation for handlebars @root context

### DISCUSSION THREAD

**Discussion by sw00sh0r** [microsoft-search/pnp-modern-search#1493] [Q&A] [answered]
Hi,

I have a question regarding the templating of the Search Result web part.
I am currently migrating the Search Result web parts in our intranet from v3.15.3 to 4.4.1 and our custom templates do not work anymore.

I believe there are some changes in the Handlebars `@root` context.
For example the primary theme in SharePoint is now addressed via
`@root.theme.palette.themePrimary`
instead of
`@root.themeVariant.palette.themePrimary`

So my guess is, that some of the other handlebar context statements are breaking my template, for example
`@root.hasPrimaryOrSecondaryResults`.

Is there any documentation about the properties inside the handlebars `root` context?
Since the handlebar-helper for logging has been omitted to make webpack work, I am no longer able to log the root context object to the dev console to inspect the object by myself.

**Comment by wobba** [comment, marked as answer]
You can use the debug template to see what the object model looks like.

**Comment by sw00sh0r** [comment]
So obvious that I would never have thought of it myself.
Thank you for the tip, that solved my problem very well!
